include/librealsense2/h/rs_device.h

```diff
@@ -188,6 +188,42 @@ const rs2_raw_data_buffer* rs2_create_flash_backup_cpp(const rs2_device* device,
 */
 const rs2_raw_data_buffer* rs2_create_flash_backup(const rs2_device* device, rs2_update_progress_callback_ptr callback, void* client_data, rs2_error** error);
 
+/**
+* Indicates whether the flash of the specified device is locked for writing
+* The device must be extendable to RS2_EXTENSION_UPDATABLE.
+* \param[in]  device        Device to check
+* \param[out] error         If non-null, receives any error that occurs during this call, otherwise, errors are ignored
+* \return true if the device flash is locked
+*/
+int rs2_is_flash_locked(const rs2_device* device, rs2_error** error);
```

Comment:
Why do you need an API and not camera info?

Reply:
it's not relevant for all devices

Reply:
Even more of reason not to make it an API...